src/mca/ptl/base/ptl_base_sendrecv.c

```diff
@@ -97,29 +98,41 @@ void pmix_ptl_base_lost_connection(pmix_peer_t *peer, pmix_status_t err)
          * participating - note that the proc would not
          * have been added to any collective tracker until
          * after it successfully connected */
-        PMIX_LIST_FOREACH(trk, &pmix_server_globals.collectives, pmix_server_trkr_t) {
+        PMIX_LIST_FOREACH_SAFE(trk, tnxt, &pmix_server_globals.collectives, pmix_server_trkr_t) {
             /* see if this proc is participating in this tracker */
             PMIX_LIST_FOREACH_SAFE(rinfo, rnext, &trk->local_cbs, pmix_server_caddy_t) {
-                if (0 != strncmp(rinfo->peer->info->pname.nspace, peer->info->pname.nspace, PMIX_MAX_NSLEN)) {
-                    continue;
-                }
-                if (rinfo->peer->info->pname.rank != peer->info->pname.rank) {
+                if (!PMIX_CHECK_PROCID(&rinfo->peer->info->pname, &peer->info->pname)) {
                     continue;
                 }
                 /* it is - adjust the count */
                 --trk->nlocal;
                 /* remove it from the list */
                 pmix_list_remove_item(&trk->local_cbs, &rinfo->super);
                 PMIX_RELEASE(rinfo);
-                /* we need to let the other participants know that this
-                 * proc has disappeared as otherwise the collective will never
-                 * complete */
+                if (0 == pmix_list_get_size(&trk->local_cbs)) {
+                    /* this tracker is complete, so release it - there
+                     * is nobody waiting for a response */
+                    pmix_list_remove_item(&pmix_server_globals.collectives, &trk->super);
+                    /* do NOT release the tracker here as the host may
+                     * have a copy they will return later. However, they
+                     * might never call back, so set a LONG timeout to
+                     * we avoid a memory leak if they don't */
+                    pmix_event_evtimer_set(pmix_globals.evbase, &trk->ev,
+                                           _timeout, trk);
+                    pmix_event_evtimer_add(&trk->ev, &tv);
+                    trk->event_active = true;
+                    break;
```

Comment:
Should you `trk->lost_connection = true` __before__ `break` ? otherwise the next `_mdxcbfunc()` will try one more time to remove the tracker from the collectives

Reply:
Grr...yep. Will add new PR to make that change

Reply:
See https://github.com/pmix/pmix/pull/959